contracts/clearing/BrokerVault.sol

```diff
@@ -83,13 +98,23 @@ contract BrokerVault is IVault2, ISettle, Ownable2Step, ReentrancyGuard {
 		INitroTypes.RecoveredVariablePart calldata candidate
 	) external {
 		uint256 channelId = NitroUtils.getChannelId(fixedPart);
+		if (performedSettlements[channelId]) {
+			revert SettlementAlreadyPerformed(channelId);
+		}
+
+		if (fixedPart.participants[1] != broker) {
+			revert BrokerNotParticipant(fixedPart.participants[1], broker);
+		}
+		address trader = fixedPart.participants[0];
 
-		require(!performedSettlements[channelId], 'Settlement already performed');
-		require(fixedPart.participants[1] == broker, 'Broker is not a participant');
-		require(
-			tradingApp.isStateTransitionValid(fixedPart, proof, candidate),
-			'Invalid state transition'
+		(bool isStateValid, string memory reason) = tradingApp.isStateTransitionValid(
+			fixedPart,
+			proof,
+			candidate
 		);
+		if (!isStateValid) {
+			revert InvalidStateTransition(reason);
+		}
```

Comment:
You can use  the `require(statementToBeTrue, CustomError())` syntax here too

Reply:
cb9be30